simpleline/render/ui_screen.py

```diff
@@ -0,0 +1,168 @@
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Author(s): Jiri Konecny <jkonecny@redhat.com>
+#
+
+from simpleline.base import App
+from simpleline.render.prompt import Prompt
+from simpleline.render.widgets import Widget
+from simpleline.utils.i18n import _
+from simpleline.event_loop.signal_handler import SignalHandler
+
+
+class UIScreen(SignalHandler):
+    """Base class representing one TUI Screen.
+
+    Shares some API with anaconda's GUI to make it easy for devs to create similar UI
+    with the familiar API.
+    """
+    # title line of the screen
+    title = u"Screen.."
+
+    def __init__(self, screen_height=25):
+        """ Constructor of the TUI screen.
+        
+        :param screen_height: height of the screen (useful for printing long widgets)
+        :type screen_height: int (the value must be bigger than 4)
+        """
+        self._screen_height = screen_height
+        self._ready = False
+
+        # list that holds the content to be printed out
+        self._window = []
+
+        # index of the page (subset of screen) shown during show_all
+        # indexing starts with 0
+        self._page = 0
+
+    def setup(self, args):
+        """Do additional setup right before this screen is used.
+
+        It is mandatory to call this ancestor method in the child class to set ready status.
+
+        :param args: arguments for the setup
+        :type args: array of values
+        :return: whether this screen should be scheduled or not
+        :rtype: bool
+        """
+        self._ready = True
+        return True
+
+    def refresh(self, args=None):
+        """Method which prepares the content desired on the screen to self._window.
+
+        :param args: optional argument passed from switch_screen calls
+        :type args: anything
+        :return: has to return True if input processing is requested, otherwise
+                 the screen will get printed and the main loop will continue
+        :rtype: True|False
+        """
+        self._window = [_(self.title), u""]
+        return True
+
+    def _print_long_widget(self, widget):
```

Comment:
We never know if the widget is too long, we just need to print a widget. So this method could be renamed to something more general like `_print_widget`.